File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/services/PermissoesServices.java
package com.example.agnciadeturismo.services;

import android.app.Activity;
import android.content.pm.PackageManager;
import android.os.Build;

import androidx.core.app.ActivityCompat;

import java.util.ArrayList;

public class PermissoesServices {

    public static void validarPermissoes(Activity activity, String[] permissoes, int requestCode){
        if(Build.VERSION.SDK_INT >= 23){
            ArrayList<String> listPermissoes = new ArrayList<>();
            for(String permissao : permissoes){
                boolean temPermissao = ActivityCompat.checkSelfPermission(activity, permissao) == PackageManager.PERMISSION_GRANTED;
                if(!temPermissao){
                    listPermissoes.add(permissao);
                }
            }

            if(!listPermissoes.isEmpty()){
                ActivityCompat.requestPermissions(activity, listPermissoes.toArray(new String[listPermissoes.size()]), requestCode);
            }
        }
    }
}

File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/presenter/view/services/PermissoesServices.kt
package com.example.agnciadeturismo.presenter.view.services

import android.app.Activity
import android.content.pm.PackageManager
import android.os.Build
import androidx.core.app.ActivityCompat

class PermissoesServices {
    companion object{
        fun validarPermissoes(activity: Activity, permissoes : Array<String>, requestCode : Int){
            if(Build.VERSION.SDK_INT >= 23){
                var listPermissoes = ArrayList<String>()
                for(permissao in permissoes){
                    var temPermissao = ActivityCompat.checkSelfPermission(activity, permissao) == PackageManager.PERMISSION_GRANTED
                    if(!temPermissao) {
                        listPermissoes.add(permissao)
                    }
                }
                if(listPermissoes.isNotEmpty()) ActivityCompat.requestPermissions(activity, listPermissoes.toTypedArray(), requestCode)
            }
        }
    }
}